ui/shared/GasSelector.qml

```diff
@@ -0,0 +1,171 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import "../imports"
+import "./"
+
+Item {
+    id: root
+    anchors.left: parent.left
+    anchors.right: parent.right
+    height: gasSlider.height + Style.current.smallPadding * 3 + txtNetworkFee.height + buttonAdvanced.height
+    property double slowestGasPrice: 0
+    property double fastestGasPrice: 100
+    property alias gasLimit: inputGasLimit.text
+    property double stepSize: ((root.fastestGasPrice - root.slowestGasPrice) / 10).toFixed(1)
+    property alias value: gasSlider.value
+    property var getGasEthValue: function () {}
+    property var getFiatValue: function () {}
+    property string defaultCurrency: "USD"
+    property string selectedGasPrice: ""
+    property string selectedGasLimit: ""
```

Comment:
It seems a bit confusing to have `gasLimit`, `selectedGasPrice`, and `selectedGasLimit`. Do you think maybe we could limit this a bit and have:
```
// property alias gasLimit: inputGasLimit.text <= REMOVE THIS
property alias selectedGasLimit: inputGasLimit.text
property alias selectedGasPrice: inputGasPrice.text

Input {
    id: inputGasLImit
    text: root.resetGasSlider() // This sets the default value of the slider, which can be overwritten by the `selectedGasLimit` property
}
```

Reply:
Yes we can totally move the default value to the UI component from Nim and make `selectedGasPrice` the single source of truth